api/src/pcapi/scripts/relocate_offers_after_oa_fix/main.py

```diff
@@ -0,0 +1,93 @@
+import argparse
+import logging
+
+from pcapi.app import app
+import pcapi.core.offerers.models as offerers_models
+import pcapi.core.offers.models as offers_models
+from pcapi.models import db
+
+
+app.app_context().push()
+logger = logging.getLogger(__name__)
+
+DATA_MODEL = ("venue_id", "correct_oa_id", "incorrect_oa_id")
+DATA = [
+    (9731, 98985, 95225),
+    (9731, 98985, 21090),
+    (9731, 98985, 95576),
+    (121823, 27031, 95193),
+    (13960, 27061, 89900),
+    (40711, 21098, 95058),
+    (5288, 55579, 89710),
+    (57776, 46293, 95323),
+]
+
+
+def relocate_offers(items: list[dict]) -> list:
+
+    ingnored_or_in_error = []
+    for item in items:
+        venue_id = item.get("venue_id")
+        correct_oa_id = item.get("correct_oa_id")
+        incorrect_oa_id = item.get("incorrect_oa_id")
+        try:
+            correct_oa = offerers_models.OffererAddress.query.filter(
+                offerers_models.OffererAddress.id == correct_oa_id
+            ).one_or_none()
+            if correct_oa is None:
+                ingnored_or_in_error.append(incorrect_oa_id)
+                logger.info("IGNORED: OA #%s not found for Venue #%s", correct_oa_id, venue_id)
+                continue
+
+            offers = (
+                offers_models.Offer.query.join(offerers_models.Venue)
+                .filter(offers_models.Offer.venueId == venue_id)
+                .filter(offers_models.Offer.offererAddressId == incorrect_oa_id)
+                .all()
+            )
+            for offer in offers:
+                offer.offererAddressId = correct_oa_id
+                offer.offererAddress = correct_oa
+                db.session.add(offer)
```

Comment:
Il n'y a pas besoin de faire les deux (assigner l'objet et l'id).
Le mieux est de laisser SQLAlchemy gerer avec l'assignation de l'objet directement.
Est-ce que l'on ne pourrait pas aussi laisser Postgresql faire la mise à jour par lot plutot qu'offre par offre ?
```suggestion
            offers = (
                offers_models.Offer.query.join(offerers_models.Venue)
                .filter(offers_models.Offer.venueId == venue_id)
                .filter(offers_models.Offer.offererAddressId == incorrect_oa_id)
                .update({offers_models.Offer.offererAddressId: correct_oa_id})
            )
```